tests/Aspire.Workload.Tests/WorkloadTestsBase.cs

```diff
@@ -0,0 +1,263 @@
+
+                int matchingEndpoints = 0;
+                var expectedEndpoints = expectedRow.Endpoints;
+
+                string[] endpointsFound =
+                    (await rowLoc.Locator("//div[@class='fluent-overflow-item']").AllAsync())
+                        .Select(async e => await e.InnerTextAsync())
+                        .Select(t => t.Result.Trim(','))
+                        .ToArray();
+                // FIXME: this could still return "+2"
+                if (endpointsFound.Length == 0)
+                {
+                    var cellText = await cellLocs[5].InnerTextAsync();
+                    endpointsFound = cellText.Trim().Split(',', StringSplitOptions.RemoveEmptyEntries);
+                }
+                if (expectedEndpoints.Length != endpointsFound.Length)
+                {
+                    // _testOutput.WriteLine($"For resource '{resourceName}, found ")
+                    // _testOutput.WriteLine($"-- expected: {expectedEndpoints.Length} found: {endpointsFound.Length}, expected: {string.Join(',', expectedEndpoints)} found: {string.Join(',', endpointsFound)} for {resourceName}");
+                    continue;
+                }
+
+                AssertEqual(expectedEndpoints.Length, endpointsFound.Length, $"#endpoints for {resourceName}");
+
+                // endpointsFound: ["foo", "https://localhost:7589/weatherforecast"]
+                foreach (var endpointFound in endpointsFound)
+                {
+                    // matchedEndpoints: ["https://localhost:7589/weatherforecast"]
+                    string[] matchedEndpoints = expectedEndpoints.Where(e => Regex.IsMatch(endpointFound, e)).ToArray();
+                    if (matchedEndpoints.Length == 0)
+                    {
+                        Assert.Fail($"Unexpected endpoint found: {endpointFound} for resource named {resourceName}. Expected endpoints: {string.Join(',', expectedEndpoints)}");
+                    }
+                    matchingEndpoints++;
+                }
+
+                AssertEqual(expectedEndpoints.Length, matchingEndpoints, $"Expected number of endpoints for {resourceName}");
+
+                // Check 'Source' column
+                AssertEqual(expectedRow.Source, await cellLocs[4].InnerTextAsync(), $"Source for {resourceName}");
+
+                foundRows.Add(expectedRow with { Endpoints = endpointsFound });
+                foundNames.Add(resourceName);
+            }
+        }
+
+        if (foundNames.Count != expectedRowsTable.Count)
+        {
+            Assert.Fail($"Expected rows not found: {string.Join(", ", expectedRowsTable.Keys.Except(foundNames))}");
+        }
+
+        return foundRows.ToArray();
+    }
+
+    // Don't fixup the prefix so it can have characters meant for testing, like spaces
+    public static string GetNewProjectId(string? prefix = null)
+        => (prefix is null ? "" : $"{prefix}_") + FixupSymbolName(Path.GetRandomFileName());
```

Comment:
GetRandomFileName returns an 8.3 file name:

[https://github.com/dotnet/runtime/blob/0b10f4ba0a9174937829ec529c197dd42a569e36/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L275-L279](https://github.com/dotnet/runtime/blob/0b10f4ba0a9174937829ec529c197dd42a569e36/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L275-L279)

We should only need to replace the `.` (but even then - why? `.` are fine in project names)

Reply:
This is originally from https://github.com/dotnet/runtime/commit/26990158d9725aa55d0af81f050a7a8e90c35f58 . But I changed this up to not fixup the whole name, but didn't get to deciding whether it could be dropped completely.

I think you are correct, and we can drop the fixup as our template generators are now to handle anything odd.

Reply:
I tested this with some more tests I have for project names, and they passed with the fixup removed.